net/respondd/src/respondd.c

```diff
@@ -292,16 +288,10 @@ static void add_provider(const char *name, const struct respondd_provider_info *
 	*pos = pentry;
 }
 
-static void load_providers(void) {
+static void load_providers(const char *path) {
 	update_time();
 
-	/* Maximum number of request types, might be made configurable in the future */
-	if (!hcreate_r(32, &htab)) {
-		perror("hcreate_r");
-		exit(EXIT_FAILURE);
-	}
-
-	DIR *dir = opendir(".");
+	DIR *dir = opendir(path);
 	if (!dir) {
 		perror("opendir");
 		exit(EXIT_FAILURE);
```

Comment:
I think it might be a good idea to ignore failures here, so we can unconditionally add certain dirs to the command line, even when no providers are installed into the dir.